crates/compiler/solve/src/solve.rs

```diff
@@ -3479,6 +3501,12 @@ fn circular_error(
     problems.push(problem);
 }
 
+/// Promotes variables at the `young_rank`, which did not escape a let-binding
+/// into a lower scope, to generalization.
```

Comment:
so "promote to generalization" means that the variable is considered for generalization?

Reply:
By "promotion" I mean when we see a variable that has a higher rank than the current scope (for example, if we solve `x = ...` where `x` is in a scope of rank 1 and `= ...` gets rank 2, and we end up with `typeof x` at rank 2), and then we mark it as generalized by giving it the special zero rank

Maybe there is a better term, or at the least we should standardize all this internally